### ISSUE TITLE
Encourage use of `[[ -n string ]]` rather than `[[ string ]]`

### ISSUE THREAD

**Issue by lollipopman** [koalaman/shellcheck#1416]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:
```sh
#!/bin/bash

if [[ $(true) ]]; then
  echo true
else
  echo false
fi 
```
#### Here's what shellcheck currently says:

No issues detected!

#### Here's what I wanted or expected to see:

```
SCXXXX: If you did intend to check that the output of the command is
not an empty string, use the longer form of the `-n` conditional expression
to clarify the intent `[[ -n $butter ]]`. However, if you intend to check the 
exit code of the command then remove the conditional expression:

  if true; then
    echo true
  else
    echo false
  fi
```
The rationale for this feature request is that one arg version of -n, i.e.
`[[ $butter ]]` is not well known and some users confuse it with how you test
the exit code of a command. Encourage people to use `-n` which is easier
to search for in documentation, and makes the code easier to grok.

**Comment by orbea** [comment]
Your suggestion is wrong, it should not warn about the lack of the `-n`, but the lack of quotes.

**Comment by lollipopman** [comment]
@orbea I agree it's a stylistic choice, but shellcheck makes other stylistic suggestions, that help make for easier to read shell code. I had a coworker make this mistake, so I thought it would be a good suggestion for shellcheck as it makes the code more explicit and easier to grok.

**Comment by orbea** [comment]
It would also cause countless incorrect warnings for correct scripts.

**Comment by koalaman** [comment]
I believe that this is a good default style to adopt due to the number of `if [ "$(grep -c ..)" ]; then ..`  and `if [ "$((RANDOM%2))" ]` style mistakes I found while examining the impact of this change.

As always, ShellCheck allows you to disable any stylistic suggestions you disagree with.
